### ISSUE TITLE
Unable to pass "empty string" variable to function when crafting proposal

### ISSUE THREAD

**Issue by adamgall** [decentdao/fractal-interface#1598]
The proposal builder marks string inputs as "required", but in solidity there is no requirement that the string be present. Meaning, an "empty string" of zero bytes is valid input to a smart contract function that takes a string.

Currently, there's no way to pass an empty string in our proposal builder.

**Comment by Da-Colon** [comment]
Question on this. What is the expected outcome here?

Say that have a function `setInputString` with a `string` argument. 

If the user leaves the value spot blank we consider that "" or undefined? 

or are you saying a user most put `""` into the space and then we parse that correctly into the transaction. currently looks like these don't have go through undefined.

**Comment by Da-Colon** [comment]
correct, currently if the value is left blank and requires a string the validation prevents if from being left blank. you are able to pass in "". but looks at the transaction for that it seems it treats it like a blank field and fails to execute version, if I'm get what you are saying should happen, I should see zero byte array.

**Comment by adamgall** [comment]
> but looks at the transaction for that it seems it treats it like a blank field and fails to execute version, if I'm get what you are saying should happen, I should see zero byte array.

?

the executed transaction results in a string of literally two double-quotes being passed into the contract function. which i believe is correct behavior! is that not what you're seeing? maybe we should just talk this out, not sure if text is conveying the message clearly.

**Comment by Da-Colon** [comment]
In Decent's UI (but could be because its not a verified contract, just one I quickly through up to test a string input. But you are right, it goes through as `""`

![Image](https://github.com/decentdao/decent-interface/assets/38386583/2cccc803-2d0d-40c5-94a7-54003be0c40e)



![Image](https://github.com/decentdao/decent-interface/assets/38386583/5546c2ec-23b0-4d37-b6c4-bfcdfaf27424)

I see what you are saying now. hmmmm yeah let's talk more about this tomorrow

**Comment by adamgall** [comment]
Same thing happens for verified contracts @Da-Colon.

Easiest for this imo is to not make string inputs required. And then make sure the "empty input" gets encoded correctly all the way through to the contracts function input as an empty string (empty byte array)

**Comment by Da-Colon** [comment]
I was thinking of making empty string (`""`) actually empty strings. The reason why its goes to save as `""` is because in our transaction build we change it to. 

![Image](https://github.com/decentdao/decent-interface/assets/38386583/19ee82cb-dac3-496a-a0ac-2e5cca0715ed)

Make makes it correct. Are you saying this should stay this way? or should empty strings act as empty string and if they want to pass 2 quotes then a user would do `""""` instead?

**Comment by adamgall** [comment]
Well it makes sense to me that whatever is typed (or not typed) into the input field should be exactly what goes into the contract function.

**Comment by Da-Colon** [comment]
that is also fair. 

So to clarify then, 

When the parameter is type 'string' then remove the requirement for the input and make it an empty string?
When its a parameter of another type keep the input required?

**Comment by adamgall** [comment]
@Da-Colon that makes sense to me yeah.
